### PR TITLE
feat: add `vitest-axe` & update frontend tests

### PR DESCRIPTION
**Pull request by jessicamcinchak**
**What is working here**
- Re-adds editor package.json script `test:silent` which CI Github workflows expect
- Swaps out `jest-axe` for `vitest-axe` & updates all a11y tests in `src/@planx/components` & beyond
  - Please note we have to install the pre-release version in order to access `toHaveNoViolations` ([see thread](https://github.com/chaance/vitest-axe/issues/13))
  - Adds `import "vitest-axe/extend-expect";` to `src/setupTests.ts`
  - Imports `axe` directly from `vitest-axe` and drops special config from `src/testUtils.ts` - does this seem okay? I don't think I've hit the SVG issue described in the code comment there yet / guessing it's old & since been resolved?
- Updates LOTS of other `jest` -> `vi` mocks along the way
- Adds per-test timeouts where necessary (see especially some public List component tests or editor Pay tests)
- Un-comments tests & `.skip()`s problematic ones so they're easier to globally search for in the final migration stretch! 

Screenshot from 2024-08-23 20-37-59

**What's not working yet / let's address in a follow up PR ?**
- All tests relying on `mockedAxios` are skipped - getting a `mockResolvedValueOnce` function not defined error (FileUploadAndLabel, Send)
- All tests relying on `vi.spyOn()` for the "default" function are skipped ([see thread](https://github.com/vitest-dev/vitest/issues/855), haven't quite sorted solution yet) (PlanningConstraints)
- Airbrake notifier tests - again a more complicated mock / spy to figure out!

Some unexpected linting stuff picked up with these changes has added extra noise to an already big PR, sorry!

### CODE REVIEW

File: editor.planx.uk/src/setupTests.ts [hunk 1]
```diff
@@ -1 +1,3 @@
 import "@testing-library/jest-dom";
+// import "@testing-library/jest-dom/vitest"; << https://stackoverflow.com/questions/77611978/invalid-chai-property-in-vitest
+import "vitest-axe/extend-expect";
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/setupTests.ts:3]: Bug here! Axe tests currently throwing `Chai property not defined` error in thread, but haven't gotten this solution working yet either

**Comment by jessicamcinchak** [comment]
[Replace Jest with Vitest in Editor to fix CI build issues](https://trello.com/c/GlouIBWM/3010-replace-jest-with-vitest-in-editor-to-fix-ci-build-issues)

File: editor.planx.uk/src/@planx/components/Pay/Editor.test.tsx [hunk 2]
```diff
@@ -107,10 +106,10 @@ describe("Pay component - Editor Modal", () => {
       expect(getByDisplayValue("myValue")).toBeInTheDocument();
     });
 
-    it("allows new values to be added", async () => {
+    it("allows new values to be added", { timeout: 20000 }, async () => {
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/Pay/Editor.test.tsx:109]: rather than a single `jest.setTimeout(2000)` at the top of the file, I've gone for per-test timeouts as it seems useful/helpful to keep a more specific pulse on which frontend interactions are slow running! 

if preferred, we still have a per-file option of `vi.setConfig({ testTimeout: 20000 })`

File: editor.planx.uk/src/@planx/components/Confirmation/Public.tsx [hunk 3]
```diff
@@ -19,11 +20,38 @@ export type Props = PublicProps<Confirmation>;
 export default function ConfirmationComponent(props: Props) {
   const [data, setData] = useState<QuestionAndResponses[]>([]);
 
-  const [sessionId, saveToEmail, $public] = useStore((state) => [
-    state.sessionId,
-    state.saveToEmail,
-    state.$public,
-  ]);
+  const [sessionId, saveToEmail, $public, passport, govUkPayment, flowName] =
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Confirmation/Public.tsx:23]: Quite a few Confirmation related changes in this PR - expecting these to go away once `dp/vite` has rebased to `main` 👍

File: editor.planx.uk/src/@planx/components/FileUploadAndLabel/Editor.test.tsx [hunk 4]
```diff
@@ -41,7 +41,7 @@ describe("FileUploadAndLabel - Editor Modal", () => {
     expect(screen.getAllByText("File")).toHaveLength(1);
   });
 
-  it.skip("allows an Editor to add multiple rules", async () => {
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/FileUploadAndLabel/Editor.test.tsx:44]: Nice catch! Worth us probably checking out all existing `skips()` once this vite + vitest migration is done and maybe opening GitHub issues for them.

File: editor.planx.uk/src/@planx/components/Pay/Editor.test.tsx (hunk 2 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/@planx/components/Pay/Editor.test.tsx:109]: This approach works well for me! Easier to come back and revisit a test at a time, than a whole suite at a time.

File: editor.planx.uk/src/@planx/components/PlanningConstraints/utils.test.ts [hunk 5]
```diff
@@ -1,26 +1,43 @@
...
+      defaultNots,
+    );
 
     expect(result).toEqual({
       intersectingConstraints: defaultIntersectingConstraints,
       nots: defaultNots,
     });
   });
 
+  // Intentional skip, not handling these edge cases yet!
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/PlanningConstraints/utils.test.ts:26]: Helpful comment!

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/tests/ServiceSettings/PublicLinks.test.tsx [hunk 6]
```diff
@@ -156,9 +157,9 @@ describe("A team without a subdomain has an offline, published service.", () =>
     });
 
     // Mocking window.location.origin
-    jest
-      .spyOn(window, "location", "get")
-      .mockReturnValue(mockWindowLocationObject);
+    vi.spyOn(window, "location", "get").mockReturnValue(
+      mockWindowLocationObject,
+    );
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Settings/tests/ServiceSettings/PublicLinks.test.tsx:162]: nit: This might be something we can put in a `beforeEach()` (not for this PR, just a reminder to come back and take a look at this!)

Also getting a few linting issues in this file, will flag with @RODO94 tomorrow or open a quick PR here.

image

File: editor.planx.uk/src/testUtils.tsx [hunk 7]
```diff
@@ -29,6 +19,15 @@ export const setup = (
...
 });
 
+// export const axe = configureAxe({
+//   rules: {
+//     // Currently, jest-axe does not correctly evaluate this rule due to an issue with jsdom
+//     // https://github.com/dequelabs/axe-core/issues/2587
+//     // To pass this test, non-decorative MUI icons should always use the 'titleAccess' prop
+//     "svg-img-alt": { enabled: false },
+//   },
+// });
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/testUtils.tsx:29]: Nice to know we don't need this - if we see an error with `titleAccess` again or failing a11y tests we can revisit.

**Review by DafyddLlyr — APPROVED** [review]
Thanks so much for this - a whole lot covered and tidied up here.

A lot of the linting issues fixed have come from some of my recent PRs, will check what's going on with Husky and try to run `git commit` from a standard directory.

File: editor.planx.uk/src/testUtils.tsx (hunk 7 above)
Comment (jessicamcinchak) [reply editor.planx.uk/src/testUtils.tsx:29]: Will properly delete in follow-up PRs rather than leave commented-out here !
